mlir/lib/Catalyst/Transforms/catalyst_to_llvm.cpp

```diff
@@ -232,6 +263,174 @@ struct PrintOpPattern : public OpConversionPattern<PrintOp> {
     }
 };
 
+// Encodes memref as LLVM struct value:
+//
+// {
+//    i64 rank,
+//    void* data,
+//    i8 type_encoding
+// }
+Value EncodeDataMemRef(Location loc, PatternRewriter &rewriter, MemRefType memrefType,
```

Comment:
How is a "data memref" different than an "opaque memref", given that both seem to consist of a rank, void*, and type encoding in the same order?

Reply:
the void* once is the data aligned (EncodeDataMemref), the other time it is the full memref descriptor, it is annoying to deal with the later when you do not need it. I could not cast it to the right type in the custom calls lib.

Reply:
Oh I see, thanks 👍